clients/src/main/java/org/apache/kafka/clients/consumer/ConsumerRecord.java

```diff
@@ -16,8 +16,9 @@
 import org.apache.kafka.common.record.TimestampType;
 
 /**
- * A key/value pair to be received from Kafka. This consists of a topic name and a partition number, from which the
- * record is being received and an offset that points to the record in a Kafka partition.
+ * A key/value pair to be received from Kafka. This also consists of a topic name and 
+ * a partition number from which the record is being received, an offset that points 
+ * to the record in a Kafka partition, and a timestamp as marked by the corresponding ProducerRecord.
```

Comment:
The timestamp is not necessarily the one in the ProducerRecord (if LogAppendTime is used), so it would be good to clarify that.